File: svc-position/src/main/java/com/beancounter/position/utils/CurrencyResolver.java
package com.beancounter.position.utils;

import com.beancounter.common.model.Currency;
import com.beancounter.common.model.Portfolio;
import com.beancounter.common.model.Position;
import com.beancounter.common.model.Trn;

public class CurrencyResolver {
  public Currency resolve(Position.In in, Portfolio portfolio, Trn trn) {

    if (in.equals(Position.In.TRADE)) {
      return trn.getTradeCurrency();
    } else if (in.equals(Position.In.PORTFOLIO)) {
      return portfolio.getCurrency();
    } else {
      return portfolio.getBase();
    }
  }
}

File: svc-position/src/main/java/com/beancounter/position/utils/CurrencyResolver.kt
package com.beancounter.position.utils

import com.beancounter.common.model.Currency
import com.beancounter.common.model.Portfolio
import com.beancounter.common.model.Position
import com.beancounter.common.model.Trn

class CurrencyResolver {
    fun resolve(`in`: Position.In, portfolio: Portfolio, trn: Trn): Currency {
        return when (`in`) {
            Position.In.TRADE -> {
                trn.tradeCurrency
            }
            Position.In.PORTFOLIO -> {
                portfolio.currency
            }
            else -> {
                portfolio.base
            }
        }
    }
}
